### Example: Isomap on Faces

• One place manifold learning is often used is in understanding the relationship between high-dimensional data points.

• A common case of high-dimensional data is images: for example, a set of images with 1,000 pixels each can be thought of as a collection of points in 1,000 dimensions – the brightness of each pixel in each image defines the coordinate in that dimension.

• Here let's apply Isomap on some faces data. We will use the Labeled Faces in the Wild dataset, which we previously saw in In-Depth: Support Vector Machines and In Depth: Principal Component Analysis.

• We begin with the standard imports:

In [2]:
%matplotlib inline
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns; sns.set()

• Running this command will download the data and cache it in your home directory for later use:

In [8]:
from sklearn.datasets import fetch_lfw_people
faces = fetch_lfw_people(min_faces_per_person=60)
faces.data.shape

(1348, 2914)

• We have 2,370 images, each with 2,914 pixels.

• In other words, the images can be thought of as data points in a 2,914-dimensional space!

• Let's quickly visualize several of these images to see what we're working with: